### PR TITLE
Update model card for Gemma

### PR DESCRIPTION
**Pull request by afafelwafi**
# What does this PR do?

Update the model card for Gemma  #36979 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/gemma.md [hunk 1]
```diff
@@ -14,31 +15,126 @@ rendered properly in your Markdown viewer.
...
-">
-<img alt="FlashAttention" src="https://img.shields.io/badge/%E2%9A%A1%EF%B8%8E%20FlashAttention-eae0c8?style=flat">
-<img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
-</div>
+[Gemma](https://goo.gle/GemmaReport) is a family of lightweight language models with pretrained and instruction-tuned variants, available in 2B and 7B parameters. The architecture is based on a transformer decoder-only design. It features attention layers with rotary positional embeddings (RoPE), SwiGLU activation functions, and RMSNorm layer normalization.
```
Comment (stevhliu) [inline docs/source/en/model_doc/gemma.md:31]:
```suggestion
[Gemma](https://huggingface.co/papers/2403.08295) is a family of lightweight language models with pretrained and instruction-tuned variants, available in 2B and 7B parameters. The architecture is based on a transformer decoder-only design. It features Multi-Query Attention, rotary positional embeddings (RoPE), GeGLU activation functions, and RMSNorm layer normalization.
```

File: docs/source/en/model_doc/gemma.md [hunk 2]
```diff
@@ -14,31 +15,126 @@ rendered properly in your Markdown viewer.
...
+
+The instruction-tuned variant was fine-tuned with supervised learning on instruction-following data, followed by reinforcement learning from human feedback (RLHF) to align the model outputs with human preferences.
+
+You can find all the original Gemma checkpoints under the [Gemma](https://huggingface.co/collections/google/gemma-release-65d5efbccdbb8c4202ec078b) release.
+
+
+> [!TIP]
+> Click on the Gemma models in the right sidebar for more examples of how to apply Gemma to different vision and language tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/gemma.md:39]:
```suggestion
> Click on the Gemma models in the right sidebar for more examples of how to apply Gemma to different language tasks.
```

File: docs/source/en/model_doc/gemma.md [hunk 3]
```diff
@@ -14,31 +15,126 @@ rendered properly in your Markdown viewer.
...
+
+The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/gemma.md:41]:
```suggestion
The example below demonstrates how to generate text with [`Pipeline`] or the [`AutoModel`] class, and from the command line.
```

File: docs/source/en/model_doc/gemma.md [hunk 4]
```diff
@@ -14,31 +15,126 @@ rendered properly in your Markdown viewer.
...
+    device_map="auto",
+    attn_implementation="sdpa"
+)
 
-Tips:
+input_text = "LLMs generate text through a process known as"
+input_ids = tokenizer(input_text, return_tensors="pt").to("cuda")
 
-- The original checkpoints can be converted using the conversion script `src/transformers/models/gemma/convert_gemma_weights_to_hf.py` 
+outputs = model.generate(**input_ids, max_new_tokens=50)
```
Comment (stevhliu) [inline docs/source/en/model_doc/gemma.md:78]:
```suggestion
outputs = model.generate(**input_ids, max_new_tokens=50, cache_implementation="static")
```

File: docs/source/en/model_doc/gemma.md [hunk 5]
```diff
@@ -14,31 +15,126 @@ rendered properly in your Markdown viewer.
...
+print(tokenizer.decode(outputs[0], skip_special_tokens=True))
+```
 
-This model was contributed by [Arthur Zucker](https://huggingface.co/ArthurZ), [Younes Belkada](https://huggingface.co/ybelkada), [Sanchit Gandhi](https://huggingface.co/sanchit-gandhi), [Pedro Cuenca](https://huggingface.co/pcuenq).
+</hfoption>
+<hfoption id="transformers-cli">
+
+```bash
+!echo -e "LLMs generate text through a process known as" | transformers-cli run --task text-generation --model google/gemma-2b --device 0
```
Comment (stevhliu) [inline docs/source/en/model_doc/gemma.md:86]:
```suggestion
echo -e "LLMs generate text through a process known as" | transformers-cli run --task text-generation --model google/gemma-2b --device 0
```

File: docs/source/en/model_doc/gemma.md [hunk 6]
```diff
@@ -14,31 +15,126 @@ rendered properly in your Markdown viewer.
...
+    quantization_config=quantization_config,
+    device_map="auto",
+    attn_implementation="sdpa"
+)
+
+input_text = "LLMs generate text through a process known as."
+input_ids = tokenizer(input_text, return_tensors="pt").to("cuda")
+outputs = model.generate(
+    **input_ids, 
+    max_new_tokens=50,
```
Comment (stevhliu) [inline docs/source/en/model_doc/gemma.md:118]:
```suggestion
    max_new_tokens=50, 
    cache_implementation="static"
```

File: docs/source/en/model_doc/gemma.md [hunk 7]
```diff
@@ -14,31 +15,126 @@ rendered properly in your Markdown viewer.
...
+visualizer = AttentionMaskVisualizer("google/gemma-2b")
+visualizer("LLMs generate text through a process known as") 
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/gemma-attn-mask.png"/>
+</div>
 
+## Notes
+The original Gemma models support standard kv-caching used in many transformer-based language models. You can use a [DynamicCache](https://github.com/huggingface/transformers/blob/main/src/transformers/cache_utils.py) instance or a tuple of tensors for past key values during generation. This makes it compatible with typical autoregressive generation workflows.
```
Comment (stevhliu) [inline docs/source/en/model_doc/gemma.md:137]: Remove the `/` at the end of the code fence below.

```suggestion
## Notes

- The original Gemma models support standard kv-caching used in many transformer-based language models. You can use use the default [`DynamicCache`] instance or a tuple of tensors for past key values during generation. This makes it compatible with typical autoregressive generation workflows.

   ```py
   import torch
   from transformers import AutoTokenizer, AutoModelForCausalLM, DynamicCache

   tokenizer = AutoTokenizer.from_pretrained("google/gemma-2b")
   model = AutoModelForCausalLM.from_pretrained(
       "google/gemma-2b",
       torch_dtype=torch.bfloat16,
       device_map="auto",
       attn_implementation="sdpa"
   )
   input_text = "LLMs generate text through a process known as"
   input_ids = tokenizer(input_text, return_tensors="pt").to("cuda")
   past_key_values = DynamicCache()
   outputs = model.generate(**input_ids, max_new_tokens=50, past_key_values=past_key_values)
   print(tokenizer.decode(outputs[0], skip_special_tokens=True))
   /```
```

**Review by stevhliu — COMMENTED** [review]
Thanks, very clean and efficient refactor!

File: docs/source/en/model_doc/gemma.md (hunk 7 above)
Comment (stevhliu) [reply docs/source/en/model_doc/gemma.md]: Remember to remove the `/` at the end here!

**Review by stevhliu — APPROVED** [review]
LGTM, thanks again! 🎉
